Add files via upload

pkubysheva opened the pull request.

pkubysheva committed:
Add files via upload

pkubysheva committed:
Add files via upload

эту задачу мы разбирали вместе с вами на паре(то,что добавила сама, закомментировала)

pkubysheva committed:
Add files via upload

Nastasia8 commented on modul5(task5).py:
```diff
@@ -0,0 +1,54 @@
+    if r-l == 1:  # если выдает 1, то возвращаем
+        it[v] = a[l]
+        return
+    m = (l+r)//2 # поиск m
```
А что такое m?)

Nastasia8 commented on modul5(task5).py:
```diff
@@ -0,0 +1,54 @@
+
+
+def build(v, l, r, it, a): #задаем переменные в функции
+    if r-l == 1:  # если выдает 1, то возвращаем
```
Что значит 1 и что мы возвращаем?)

Nastasia8 commented on modul5(task5).py:
```diff
@@ -0,0 +1,54 @@
+from math import gcd
+
+
+def build(v, l, r, it, a): #задаем переменные в функции
```
Для большей убедительности, лучше описывать, что каждая переменная из себя представляет.

Nastasia8 commented on modul5(task5).py:
```diff
@@ -0,0 +1,54 @@
+
+
+def getNod(v, l, r, it, ql, qr): #задаем переменные в функции
+    if ql <= l and qr >= r: # если
```
?

Nastasia8 commented on modul5(task5).py:
```diff
@@ -0,0 +1,54 @@
+def getNod(v, l, r, it, ql, qr): #задаем переменные в функции
+    if ql <= l and qr >= r: # если
+        return it[v]
+    if ql >= r or qr <= l: # если слева больше, чем справа или наоборот, то возвращаем 0
```
ну, очень туманно) Вы должны описывать к чему  приводит выполнение данного условия и как оно достигается))

Nastasia8 commented on modul5(task5).py:
```diff
@@ -0,0 +1,54 @@
+    if ql <= l and qr >= r: # если
+        return it[v]
+    if ql >= r or qr <= l: # если слева больше, чем справа или наоборот, то возвращаем 0
+        return 0
```
И почему же 0?))) А не 1, 2, -1, 10000 и т.д.?)

Nastasia8 commented on modul5(task5).py:
```diff
@@ -0,0 +1,54 @@
+
+def main(): #создаем функции
+    n = int(input()) #ввод по условию
+    it = [0]*(4*n) #ввод по условию
```
Вот здесь мы выделяем память под хранение дерева отрезков.

Nastasia8 commented on modul5(task5).py:
```diff
@@ -0,0 +1,54 @@
+    res = []
+    while q != 0: #пока не равно нулю выполняем:
+        type_q, l, r = map(str, input().split())
+        if type_q == "s": # если это элементы левой или правой границы
```
странный комментарий)
type_q - хранит тип запроса: либо s, либо u
Если это s-запрос, то вычисляем NOD на отрезке, иначе, это u-запрос, с помощью которого обновляем значение элемента по его индексу.

pkubysheva committed:
Add files via upload

эти задачи уже загружала месяца 2 назад
Вы ,наверное, их пропустили, поэтому вот)

pkubysheva committed:
Add files via upload

эти задачи уже загружала когда-то
Вы их наверное пропустили, поэтому вот)
